core/node/provider.go

```diff
@@ -159,32 +155,65 @@ func OfflineProviders() fx.Option {
 	return fx.Provide(provider.NewNoopProvider)
 }
 
-func newProvidingStrategy(onlyPinned, onlyRoots bool) interface{} {
+func mfsProvider(mfsRoot *mfs.Root, fetcher fetcher.Factory) provider.KeyChanFunc {
+	return func(ctx context.Context) (<-chan cid.Cid, error) {
+		rootNode, err := mfsRoot.GetDirectory().GetNode()
+		if err != nil {
+			return nil, fmt.Errorf("error loading mfs root, cannot provide MFS: %w", err)
+		}
+
+		kcf := provider.NewDAGProvider(rootNode.Cid(), fetcher)
+		return kcf(ctx)
+	}
+
+}
+
+func mfsRootProvider(mfsRoot *mfs.Root) provider.KeyChanFunc {
+	return func(ctx context.Context) (<-chan cid.Cid, error) {
+		rootNode, err := mfsRoot.GetDirectory().GetNode()
+		if err != nil {
+			return nil, fmt.Errorf("error loading mfs root, cannot provide MFS: %w", err)
+		}
+		ch := make(chan cid.Cid, 1)
+		ch <- rootNode.Cid()
+		close(ch)
+		return ch, nil
+	}
+}
+
+func newProvidingStrategy(strategy string) interface{} {
 	type input struct {
 		fx.In
-		Pinner      pin.Pinner
-		Blockstore  blockstore.Blockstore
-		IPLDFetcher fetcher.Factory `name:"ipldFetcher"`
+		Pinner               pin.Pinner
+		Blockstore           blockstore.Blockstore
+		IPLDFetcher          fetcher.Factory `name:"ipldFetcher"`
+		OfflineUnixFSFetcher fetcher.Factory `name:"offlineUnixfsFetcher"`
```

Comment:
used for mfs assuming that only unixfs can make it into mfs...

Reply:
Yes, MFS is UnixFS only (raw or dag-pb). It is something one would mount as local filesystem.
we added extra check for that in https://github.com/ipfs/kubo/pull/10701/files